src/main/java/tatracker/logic/parser/ParserUtil.java

```diff
@@ -288,4 +288,19 @@ public static Tab parseTabName(String tabName) throws ParseException {
         }
         return Tab.getTab(trimmedType);
     }
+
+    public static int parseRate(String rate) throws ParseException {
+        requireNonNull(rate);
+        String trimmedRate = rate.trim();
+         try {
+             int parsedRate = Integer.parseInt(trimmedRate);
+             if (parsedRate <= 0) {
+                 throw new ParseException("RATE must be a positive integer");
+             }
+             System.out.println("Parse Rate:" + parsedRate);
+             return parsedRate;
+         } catch (NumberFormatException e) {
+             throw new ParseException("RATE must be a positive integer");
+        }
+    }
 }
```

Comment:
There's a parse unsigned integer function alr. Can reuse that in this function, but catch and throw another exception with the new messages.

Also careful with debugging

Reply:
I can parse +1 and it will be valid.